models/lstm/include/LSTM.h

```diff
@@ -0,0 +1,142 @@
+#ifndef LSTM_H
+#define LSTM_H
+
+#include "all.h"
+#include "lstm_fluxes.h"
+#include "lstm_params.h"
+#include "lstm_config.h"
+#include "lstm_state.h"
+#include <unordered_map>
+
+#ifdef NGEN_LSTM_TORCH_LIB_ACTIVE
```

Comment:
Can we move this macro to the top of the file?  again, no need to include headers even if we don't have torch?